Source/Core/UICommon/GameFileCache.cpp

```diff
@@ -102,21 +105,27 @@ bool GameFileCache::Update(const std::vector<std::string>& all_game_paths)
       }
       else
       {
+        if (game_removed_from_cache)
+          game_removed_from_cache((*it)->GetFilePath());
+
         cache_changed = true;
         --end;
         *it = std::move(*end);
+        m_cached_files.pop_back();
       }
     }
-    m_cached_files.erase(it, m_cached_files.end());
   }
 
   // Now that the previous loop has run, game_paths only contains paths that
   // aren't in m_cached_files, so we simply add all of them to m_cached_files.
-  for (const auto& path : game_paths)
+  for (const std::string& path : game_paths)
```

Comment:
Why change it here when you kept ``auto`` in the other loops?

Reply:
Which other loops? I can only find one loop in this file that uses auto, and that auto represents a longer type than std::string. Though I guess I could change that one too...